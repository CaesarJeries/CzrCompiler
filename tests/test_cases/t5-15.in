int f1() {
    print("a\n");
    return 10;
}

byte f2(){
    print("b\n");
    return 11b;
}

bool f3() {
    print("c\n");
    return false;
}

void foo(int a1, int a2, int a3) {
    printi(a1);
    printi(a2);
    printi(a3);
}

void bar(bool c1, int a2, int a3) {
    if (c1) printi(1); else printi(0);
    printi(a2);
    printi(a3);
}

void main(){
    foo(f1(),0,f2());
    print("\n");
    foo(f2(),f1(),10);
    print("\n");
    foo(f1(),f2(),f2());
    print("\n");
    bar(f3(),f2(),f1());
    print("\n");
}