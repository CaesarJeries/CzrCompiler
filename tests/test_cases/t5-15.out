Loaded: ./exceptions.s
a
b
10011
b
a
111010
a
b
b
101111
c
b
a
01110
